lib/kb_ObjectInfo/CreateAssemblyReport.py

```diff
@@ -0,0 +1,139 @@
+                assem_list.append(" ")
+                
+            # Add the row to the list that will be returned
+            this_list.extend([assem_list])
+            
+        return (header,this_list)
+        
+    def assembly_dnabases(self,obj_list):
+        header = "DNA Composition"
+        this_list = [["Assembly Name","Total DNA Bases",
+                        "A Count","A Percent","C Count","C Percent",
+                        "G Count","G Percent","T Count","T Percent"]]
+ 
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            if 'info' in assembly:
+                name = assembly['info'][1]
+            dna_size = 1.0
+            pct = 1.00
+            if 'dna_size' in assembly['data']:
+                dna_size = assembly['data']['dna_size']
+            
+            assem_list = [name,str(dna_size)]
+            for base in ["A","C","G","T"]:
+                pct = round(100 * assembly['data']['base_counts'][base] / dna_size,2)
+                assem_list.append(str(assembly['data']['base_counts'][base]))
+                assem_list.append(str(pct))
+                
+            this_list.extend([assem_list])
+            
+        return(header,this_list)
+                
+    def assembly_contigs(self,obj_list):
+        header = "Contigs in the Assembly"
+        this_list= [["Assembly Name","Contig Name","Length","GC content","Number of Ns","Contig ID","Description"]]
+        
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            if 'info' in assembly:
+                name = assembly['info'][1]
+ 
+            if 'contigs' in assembly['data']:
+                myContig = assembly['data']['contigs']
+                for ctg in myContig:
+                    list = ['length', 'gc_content', 'Ncount', 'contig_id', 'description']
+                    ctg_list = [name,ctg]
+                
+                    for item in list:
+                        if item in myContig[ctg]:
+                            ctg_list.append(format(myContig[ctg][item]))
+                        else:
+                            ctg_list.append("")
+
+                    this_list.append(ctg_list)
+                        
+        return(header,this_list)
+        
+                        
+    def assembly_dna(self,obj_list,scratch):
+        header = "Contig FastA files found in the download files."
+        fasta_list = []
+        dna_string = ""
+        cf = CreateFasta(self.config)
+        
+        for assembly in obj_list['data']:
+            name = "Assembly Data Object"
+            input_ref = ''
+            if 'info' in assembly:
+                name = assembly['info'][1]
+                input_ref = str(assembly['info'][6]) + "/" + str(assembly['info'][0]) + "/" + str(assembly['info'][4])
+
+                fasta_list = cf.get_assembly_sequence(input_ref)
+                report_path = os.path.join(scratch, name + '.fna')
```

Comment:
I'm going to push back on this one. 

If I was using some random name that the user wanted, then this would be a very valid point. Instead, I am using the data object name to assign the file name. This naming method is used in the kb_staging_exporter so I assume that it has been tested and blessed.

Reply:
Well, I only thought it was an issue because the user can always change the object name in the Narrative. When I just tested this, the only character that slipped through the Narrative ui object renaming renaming tool was the pipe | character. But it looks like DIE replaces that with an underscore anyway, so this is all fine.